dask/array/core.py

```diff
@@ -1047,6 +1047,19 @@ def _set_chunks(self, chunks):
     def __len__(self):
         return sum(self.chunks[0])
 
+    def __array_ufunc__(self, numpy_ufunc, method, *inputs, **kwargs):
+        import pdb; pdb.set_trace()
+        from . import ufunc
+        da_ufunc = getattr(ufunc, numpy_ufunc.__name__)
```

Comment:
You need to return `NotImplemented` here for unrecognized types, e.g., [something along these lines](https://github.com/numpy/numpy/blob/d93a5dd6b15fcc10377c484247f84ad80ccd3b66/numpy/lib/tests/test_mixins.py#L27-L34). Otherwise, dask will take precedence over everything else, even libraries like xarray.

(There are also some subtleties around output types and the `out` argument that I would recommend copying from that example, though possibly it would make sense to always return `NotImplemented` if `out` is set.)